apps/authentication/backends/base.py

```diff
@@ -44,14 +43,22 @@ def allow_authenticate(self, user=None, username=None):
             # 特殊值 None 表示没有限制
             return True
         backend_name = self.__class__.__name__
-        allowed_backend_names = [path.split('.')[-1] for path in allowed_backend_paths]
+        allowed_backend_names = [path.split('.drf.py')[-1] for path in allowed_backend_paths]
         allow = backend_name in allowed_backend_names
         if not allow:
             info = 'User {} skip authentication backend {}, because it not in {}'
             info = info.format(username, backend_name, ','.join(allowed_backend_names))
             logger.info(info)
         return allow
 
+    def get_user(self, user_id):
+        """ 三方用户认证成功后 request.user 赋值时会调用 backend 的当前方法获取用户 """
+        try:
+            user = UserModel._default_manager.get(pk=user_id)
+        except UserModel.DoesNotExist:
+            return None
+        return user if self.user_can_authenticate(user) else None
+
 
 class JMSModelBackend(JMSBaseAuthBackend, ModelBackend):
     pass
```

Comment:
The code provided does not contain any irregularities, potential issues, or optimization suggestions. It seems like a standard Django/JWT authentication setup implemented in Python with some optional extensions such as custom error messages and logging. However, there might be slight changes depending on how you customize the code for your specific use case, but those should remain unchanged from this example.